app/assets/javascripts/oxalis/view/user_scripts_modal.js

```diff
@@ -14,14 +16,47 @@ class UserScriptsModalView extends React.PureComponent {
 
   state = {
     code: "",
+    isCodeChanged: false,
+    scripts: [],
+    // Needs to be undefined so the placeholder is displayed in the beginning
+    selectedScript: undefined,
+    isLoading: true,
   };
 
+  componentWillMount() {
+    Request.receiveJSON("/api/scripts").then(scripts => {
+      this.setState({ isLoading: false });
+      if (scripts.length) {
+        this.setState({ scripts });
+      }
+    });
+  }
+
   handleCodeChange = (event: SyntheticInputEvent) => {
     this.setState({
       code: event.target.value,
+      isCodeChanged: true,
     });
   };
 
+  handleScriptChange = async (scriptId: string) => {
+    const script = this.state.scripts.find(s => s.id === scriptId);
+    if (script == null) return;
+    if (
+      !this.state.isCodeChanged ||
+      confirm("This will replace the code you've written. Continue?")
```

Comment:
Please move the text to `message.js` file. Also, please spell out `have`.

I would love to see antd's  `Modal.Confirm` here. https://ant.design/components/modal/